### PR TITLE
[4.0] Router fix - layouts

### PR DESCRIPTION
**Pull request by alexandraciobica**
Pull Request for Issues #20958, #20485.

### Summary of Changes
Removed the line that was setting the lookup for the view without layout because it was overwriting the `Itemid` of the menu item that doesn't have a layout with the `Itemid` of the menu item that has a layout.

### Result
The router works for views with layouts.

### Testing Instructions
1. Create a menu item for the "User profile", another one for "Edit user profile"
2. Go to "Edit user profile", click cancel.
Result: Redirect to the "User profile" page

Note: Talked with @Hackwar, @bembelimen about these changes.

### CODE REVIEW

File: libraries/src/Component/Router/Rules/MenuRules.php [hunk 1]
```diff
@@ -212,7 +212,8 @@ protected function buildLookup($language = '*')
 
 					$layout = '';
 
-					if (isset($item->query['layout']))
+					// Checking if layout is default just to be safe
+					if (isset($item->query['layout']) || $item->query['layout'] == 'default')
```
Comment (SharkyKZ) [inline libraries/src/Component/Router/Rules/MenuRules.php]: This change is not needed and will cause undefined index notices.

**Comment by SharkyKZ** [comment]
Can this be done against 3.x? Or is this not B/C safe?

**Comment by alexandraciobica** [comment]
@SharkyKZ I tested it and it doesn't work for 3.x.

**Comment by ggppdk** [comment]
@SharkyKZ 
@alexandraciobica 


That line is there for a purpose, to remove it is not that simple, it will effect many views and components

In J3 there are 2 places to remove it , but it is problematic
#19516

Please see description in a more complex PR here:
#19945

for a more complete solution , which might be good to be done for J4

**Comment by Hackwar** [comment]
I disagree with @ggppdk here. I have to admit that I did not test this intensively, but I went through the code with @alexandraciobica and that one line is wrong there. We are building a lookup with the view and the layout as key. That additional line creates a "fallback" for links where the layout might not match, but the view does. That is usefull for situations where variable content might be displayed differently, like with the category view, but it is pretty much useless for cases like the profile view in com_users. So we want to have the fallback in 246 for the variable content, but not here in line 259 for the fixed views, because with that "fallback", we would overwrite the original view with the more specific layout. That would specifically mean that the menu item for editing a profile would overwrite the menu item for viewing a profile if the later was processed before the former. Or a "thank you" menu item of a contact form would overwrite the original menu item for the contact form.

Long story short: It seems as if we really just need to remove that one line to fix this issue.

**Comment by ggppdk** [comment]
@Hackwar 

so i understand that the points mentioned in J3 PRs are no longer valid in J4
please post a successful test since you tested it
i admit i have not tested this PR

**Comment by fancyFranci** [comment]
I have tested this item :white_check_mark: successfully on 45356411c901721e77a8e58561ba46f4d6b672ea<br /><br />**Before patch:**
index.php?option=com_users&view=profile **=>**
/joomla4/index.php/edit-profile
index.php?option=com_users&view=profile&layout=edit **=>**
/joomla4/index.php/edit-profile

**After patch:**
index.php?option=com_users&view=profile **=>**
/joomla4/index.php/profile
index.php?option=com_users&view=profile&layout=edit **=>**
/joomla4/index.php/edit-profile

In **Joomla3** it works with and without this patch. It just appends '&layout=edit' to the profile-url. That's not right, but it works.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21187">issues.joomla.org/tracker/joomla-cms/21187</a>.</sub>

**Comment by infograf768** [comment]
Whatever you think, please test on a multilingual site...

**Comment by fancyFranci** [comment]
I have tested this item :red_circle: unsuccessfully on 45356411c901721e77a8e58561ba46f4d6b672ea<br /><br />Ok ok... on a multilanguage site the patch has no effect.
Coming from profile menu item you go back to profile by clicking 'Cancel' in the editing form.
Edit-profile as menu item stays on the form after clicking 'Cancel'.

Always:
index.php?option=com_users&view=profile **=>**
/joomla4ml/index.php/en/profile
index.php?option=com_users&view=profile&layout=edit **=>**
/joomla4ml/index.php/en/profile?layout=edit<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21187">issues.joomla.org/tracker/joomla-cms/21187</a>.</sub>

**Comment by ceus1984** [comment]
I have tested this item :white_check_mark: successfully on 45356411c901721e77a8e58561ba46f4d6b672ea<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21187">issues.joomla.org/tracker/joomla-cms/21187</a>.</sub>

**Comment by ghost** [comment]
@ceus1984 
Can you describe, where the cancel button is located? I've only found a close button.

**Comment by webnet-assmann** [comment]
I have tested this item :white_check_mark: successfully on 45356411c901721e77a8e58561ba46f4d6b672ea<br /><br />Worked also on an multilingual site<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21187">issues.joomla.org/tracker/joomla-cms/21187</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by laoneo** [comment]
Thanks for the pr.
